influxdb/influxdb-neb.lua

```diff
@@ -145,20 +145,22 @@ function EventQueue:add(e)
     end
     -- message format : <measurement>[,<tag-key>=<tag-value>...]
     --  <field-key>=<field-value>[,<field2-key>=<field2-value>...] [unix-nano-timestamp]
+    -- some characters [ ,=] must be escaped, let's replace them by _ for better handling
     -- consider space in service_description as a separator for an item tag
     local item = ""
-    if string.find(service_description, " ") then
-        item = ",item=" .. string.gsub(service_description, ".* ", "")
-        service_description = string.gsub(service_description, " .*", "")
+    if string.find(service_description, " [^ ]+$") then
+        item = ",item=" .. string.gsub(string.gsub(service_description, ".* ", "", 1), "[ ,=]+" ,"_")
+        service_description = string.gsub(service_description, " +[^ ]+$", "", 1)
```

Comment:
Maybe it would have been interesting to set the "_" customizable. We could use the QueueEvent.new() method for this purpose by setting "_" by default and maybe change the code a little so that we can change it in the GUI.

Another point is string.gsub(string.gsub(...)). I don't like the idea of regex parsing in another regex parsing, it becomes heavy... I don't have enough time to find a better solution now. Maybe a cache could be used with the service description given as key and what you want as replacement in values.
It would be interesting to dig into it...

Anyway, thank you for this patch.

Reply:
I'm not sure to fully understand what you want to be customizable @bouda1 :)

Reply:
Just a suggestion! You force the replacement of [=,] by '_'. I propose to set '_' in a variable and expose it so that it can be changed by another customer. That's all.

Reply:
Understood :)
You're right, we could make the replacement a tunable :+1:
Thank you for having merged this patch !